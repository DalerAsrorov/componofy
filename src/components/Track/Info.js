import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Chip from '@material-ui/core/Chip';
import { purple, red } from '@material-ui/core/colors';

import './Track.css';

const styles = (theme) => ({
  root: theme.mixins.gutters({
    paddingTop: 6,
    paddingBottom: 6,
  }),

  popularity: {
    // padding: `${theme.spacing.unit}px 0`
    padding: '0',
  },

  chip: {
    color: red[50],
    backgroundColor: purple['A200'],
    height: '20px',
    padding: '2px',
    marginTop: '5px',
  },
});

const createTypographyLink = (content, variant, href) => (
  <Typography
    className="link-default"
    variant={variant}
    component="a"
    href={href}
    target="__blank"
  >
    {content}
  </Typography>
);

const Info = (props) => {
  let popularityChip;

  if (props.isPopular) {
    const labelComponent = (
      <Typography variant="caption" color="textPrimary">
        Popular
      </Typography>
    );
    popularityChip = (
      <div className={props.classes.popularity}>
        <Chip label={labelComponent} className={props.classes.chip} />
      </div>
    );
  }

  return (
    <div className={props.classes.root}>
      {createTypographyLink(props.trackName, 'subtitle1', props.trackUrl)}
      {createTypographyLink(props.artistName, 'body1', props.artistUrl)}
      {createTypographyLink(props.albumName, 'caption', props.albumUrl)}
      {popularityChip}
    </div>
  );
};

Info.propTypes = {
  trackName: PropTypes.string.isRequired,
  trackUrl: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired,
  artistUrl: PropTypes.string.isRequired,
  albumName: PropTypes.string.isRequired,
  albumUrl: PropTypes.string.isRequired,
  isPopular: PropTypes.bool.isRequired,
};

export default withStyles(styles)(Info);
