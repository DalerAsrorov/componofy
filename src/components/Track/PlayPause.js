import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton';
import { PlayArrow, Pause } from '@material-ui/icons';
import { withMediaProps } from 'react-media-player';

class PlayPause extends PureComponent {
  static propTypes = {
    media: PropTypes.object.isRequired,
    color: PropTypes.string,
  };

  _handlePlayPause = () => {
    this.props.media.playPause();
  };

  render() {
    const { media, color } = this.props;
    let playPauseIcon = <Pause />;

    if (!media.isPlaying) {
      playPauseIcon = <PlayArrow />;
    }

    return (
      <IconButton color={color} onClick={this._handlePlayPause}>
        {playPauseIcon}
      </IconButton>
    );
  }
}

export default withMediaProps(PlayPause);
