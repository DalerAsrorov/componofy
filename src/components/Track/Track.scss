.link-default {
  text-decoration: none;
}
