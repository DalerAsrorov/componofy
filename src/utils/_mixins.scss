@mixin vertical-align($position: relative) {
  position: $position;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}
