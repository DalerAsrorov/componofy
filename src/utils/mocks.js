import { swapKeysAndValues } from './helpers';

export const MOCK_EXTERNAL_URL = {
  spotify: 'test_external_url'
};

export const ROUTE_INDEX_MAP = {
  0: '/app',
  1: '/app/public',
  2: '/app/componofy'
};

export const MOCK_IMAGE = {
  url: 'test image url',
  height: 500,
  width: 100
};

export const DEFAULT_INDEX = 0;

export const MOCK_NAVIGATION = {
  index: 0,
  currentPage: ROUTE_INDEX_MAP[DEFAULT_INDEX],
  nextIndex: 1,
  nextPage: ROUTE_INDEX_MAP[DEFAULT_INDEX + 1],
  indexToRouteMap: ROUTE_INDEX_MAP,
  routeToIndexMap: swapKeysAndValues(ROUTE_INDEX_MAP, parseInt),
  numberOfPages: 3
};

export const MOCK_ARTIST = {
  id: '324',
  name: 'Test Artist',
  external_urls: MOCK_EXTERNAL_URL
};

export const MOCK_ALBUM = {
  artists: [MOCK_ARTIST],
  name: 'Test album',
  external_urls: MOCK_EXTERNAL_URL,
  album_type: 'test type',
  images: [MOCK_IMAGE, MOCK_IMAGE]
};

export const MOCK_TRACK = {
  id: '1234',
  artists: [MOCK_ARTIST],
  external_urls: MOCK_EXTERNAL_URL,
  track_number: 1,
  duration_ms: 12323,
  popularity: 78,
  album: MOCK_ALBUM
};

export const MOCK_TRACKS = [MOCK_TRACK, MOCK_TRACK, MOCK_TRACK];

export const MOCK_PLAYLIST = {
  id: '8752',
  href: 'testLink',
  name: 'Artist 1',
  external_urls: MOCK_EXTERNAL_URL,
  tracks: {
    list: MOCK_TRACKS
  },
  owner: {
    id: 'testOwner',
    type: 'testType'
  },
  images: [MOCK_IMAGE],
  type: 'testType',
  public: true,
  isOpen: false
};

export const MOCK_PLAYLISTS = [
  {
    id: '8752',
    href: 'testLink',
    name: 'Artist 1',
    external_urls: MOCK_EXTERNAL_URL,
    tracks: {
      list: []
    },
    owner: {
      id: 'testOwner',
      type: 'testType'
    },
    images: [MOCK_IMAGE],
    type: 'testType',
    public: true,
    isOpen: false
  },
  {
    id: '6547',
    href: 'testLink',
    name: 'Artist 1',
    external_urls: MOCK_EXTERNAL_URL,
    tracks: {
      list: []
    },
    owner: {
      id: 'testOwner',
      type: 'testType'
    },
    images: [MOCK_IMAGE],
    type: 'testType',
    public: true,
    isOpen: false
  },
  {
    id: '4234',
    href: 'testLink',
    name: 'Artist 1',
    external_urls: MOCK_EXTERNAL_URL,
    tracks: {
      list: []
    },
    owner: {
      id: 'testOwner',
      type: 'testType'
    },
    images: [MOCK_IMAGE],
    type: 'testType',
    public: true,
    isOpen: false
  }
];
