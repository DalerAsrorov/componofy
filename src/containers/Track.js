import { withRouter } from 'react-router-dom';
import { connectStream } from '../connectPage';
import Track from '../components/Track';

export default connectStream(withRouter(Track));
