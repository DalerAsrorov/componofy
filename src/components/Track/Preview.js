import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import PlayPause from './PlayPause';
import { Media, Player } from 'react-media-player';

const styles = theme => ({
    mediaPlayer: {
        display: 'none'
    },

    media: {
        textAlign: 'center'
    }
});

const Preview = props => (
    <Media>
        <div className={props.classes.media}>
            <div className={props.classes.mediaPlayer}>
                <Player src={props.url} vendor="audio" />
            </div>
            <PlayPause color="secondary" />
        </div>
    </Media>
);

Preview.propTypes = {
    classes: PropTypes.object.isRequired,
    url: PropTypes.string.isRequired,
    playButtonColor: PropTypes.string
};

export default withStyles(styles)(Preview);
