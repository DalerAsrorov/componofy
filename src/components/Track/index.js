import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Checkbox from 'material-ui/Checkbox';
import MaterialList, {
    ListItem,
    ListItemIcon,
    ListItemText
} from 'material-ui/List';
import { withStyles } from 'material-ui/styles';
import { head } from 'ramda';
import { TRACK_PROPTYPE, PLAYLIST_PROPTYPE } from '../../utils/constants';

import Info from './Info';

const styles = theme => ({
    checkmark: {
        color: theme.palette.grey[600]
    }
});

class Track extends PureComponent {
    static propTypes = {
        addPlaylistTrackToFinal: PropTypes.func.isRequired,
        track: TRACK_PROPTYPE.isRequired,
        playlist: PLAYLIST_PROPTYPE
    };

    state = {
        // TODO: use mapStateToProps from redux
        // to check if track is in the queue of
        // added tracks. If it is, then mark
        // the prop "isAdded" as true, otherwise
        // false. Once that is done, this should
        // be moved to props of boolean type.
        isAdded: false
    };

    _handleChecked = event => {
        const { isAdded } = this.state;
        const { track, playlist, addPlaylistTrackToFinal } = this.props;

        if (playlist) {
            addPlaylistTrackToFinal(track, playlist);
        }

        this.setState({ isAdded: !isAdded });
    };

    render() {
        const { isAdded } = this.state;
        const { track, classes } = this.props;

        const {
            id: trackID,
            artists,
            name: trackName,
            album: { name: albumName, external_urls: { spotify: albumUrl } },
            external_urls: { spotify: trackUrl },
            popularity
        } = track;
        const {
            name: artistName,
            external_urls: { spotify: artistUrl }
        } = head(artists);

        const isPopular = popularity >= 70;

        return (
            <ListItem divider>
                <ListItemIcon>
                    <Checkbox
                        className={classes.checkmark}
                        onClick={this._handleChecked}
                        checked={isAdded}
                    />
                </ListItemIcon>
                <ListItemText
                    primary={
                        <Info
                            trackName={trackName}
                            trackUrl={trackUrl}
                            artistName={artistName}
                            artistUrl={artistUrl}
                            albumName={albumName}
                            albumUrl={albumUrl}
                            isPopular={isPopular}
                        />
                    }
                />
            </ListItem>
        );
    }
}

export default withStyles(styles)(Track);
