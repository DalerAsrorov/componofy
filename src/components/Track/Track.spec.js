import React from 'react';
import Tabs, { Tab } from 'material-ui/Tabs';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createShallow, createMount } from 'material-ui/test-utils';
import { MOCK_TRACK, MOCK_PLAYLIST, MOCK_NAVIGATION } from '../../utils/mocks';
import Track from './';

Enzyme.configure({ adapter: new Adapter() });

const requiredProps = {
  removePlaylistTrackFromFinal: jest.fn(),
  playlistContainsThisTrack: false,
  addPlaylistTrackToFinal: jest.fn(),
  addErrorToApp: jest.fn(),
  index: 1,
  track: MOCK_TRACK,
  playlist: MOCK_PLAYLIST,
  navigation: MOCK_NAVIGATION
};

/**
 * NOTE: Testing this causes a lot of errors
 * will need to re-visit on how to test this huge component
 * properly.
 */
describe('Track component', () => {
  let mount, shallow;

  beforeEach(() => {
    mount = createMount();
    shallow = createShallow();
  });

  it('renders with required props', () => {
    let wrapper = shallow(<Track {...requiredProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
