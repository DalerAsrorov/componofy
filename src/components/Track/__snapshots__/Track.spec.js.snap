// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track component renders with required props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Track)
    addErrorToApp={[Function]}
    addPlaylistTrackToFinal={[Function]}
    index={1}
    navigation={
      Object {
        "currentPage": "/app",
        "index": 0,
        "indexToRouteMap": Object {
          "0": "/app",
          "1": "/app/public",
          "2": "/app/componofy",
        },
        "nextIndex": 1,
        "nextPage": "/app/public",
        "numberOfPages": 3,
        "routeToIndexMap": Object {
          "/app": 0,
          "/app/componofy": 2,
          "/app/public": 1,
        },
      }
    }
    playlist={
      Object {
        "external_urls": Object {
          "spotify": "test_external_url",
        },
        "href": "testLink",
        "id": "8752",
        "images": Array [
          Object {
            "height": 500,
            "url": "test image url",
            "width": 100,
          },
        ],
        "isOpen": false,
        "name": "Artist 1",
        "owner": Object {
          "id": "testOwner",
          "type": "testType",
        },
        "public": true,
        "tracks": Object {
          "list": Array [
            Object {
              "album": Object {
                "album_type": "test type",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "images": Array [
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                ],
                "name": "Test album",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "id": "324",
                  "name": "Test Artist",
                },
              ],
              "duration_ms": 12323,
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "1234",
              "popularity": 78,
              "track_number": 1,
            },
            Object {
              "album": Object {
                "album_type": "test type",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "images": Array [
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                ],
                "name": "Test album",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "id": "324",
                  "name": "Test Artist",
                },
              ],
              "duration_ms": 12323,
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "1234",
              "popularity": 78,
              "track_number": 1,
            },
            Object {
              "album": Object {
                "album_type": "test type",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "images": Array [
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                ],
                "name": "Test album",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "id": "324",
                  "name": "Test Artist",
                },
              ],
              "duration_ms": 12323,
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "1234",
              "popularity": 78,
              "track_number": 1,
            },
          ],
        },
        "type": "testType",
      }
    }
    playlistContainsThisTrack={false}
    removePlaylistTrackFromFinal={[Function]}
    track={
      Object {
        "album": Object {
          "album_type": "test type",
          "artists": Array [
            Object {
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "324",
              "name": "Test Artist",
            },
          ],
          "external_urls": Object {
            "spotify": "test_external_url",
          },
          "images": Array [
            Object {
              "height": 500,
              "url": "test image url",
              "width": 100,
            },
            Object {
              "height": 500,
              "url": "test image url",
              "width": 100,
            },
          ],
          "name": "Test album",
        },
        "artists": Array [
          Object {
            "external_urls": Object {
              "spotify": "test_external_url",
            },
            "id": "324",
            "name": "Test Artist",
          },
        ],
        "duration_ms": 12323,
        "external_urls": Object {
          "spotify": "test_external_url",
        },
        "id": "1234",
        "popularity": 78,
        "track_number": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addErrorToApp": [Function],
      "addPlaylistTrackToFinal": [Function],
      "classes": Object {
        "mediaPlayer": "Track-mediaPlayer-4",
        "preview": "Track-preview-3",
        "trackInfo": "Track-trackInfo-2",
        "trackInfoContainer": "Track-trackInfoContainer-1",
      },
      "index": 1,
      "navigation": Object {
        "currentPage": "/app",
        "index": 0,
        "indexToRouteMap": Object {
          "0": "/app",
          "1": "/app/public",
          "2": "/app/componofy",
        },
        "nextIndex": 1,
        "nextPage": "/app/public",
        "numberOfPages": 3,
        "routeToIndexMap": Object {
          "/app": 0,
          "/app/componofy": 2,
          "/app/public": 1,
        },
      },
      "playlist": Object {
        "external_urls": Object {
          "spotify": "test_external_url",
        },
        "href": "testLink",
        "id": "8752",
        "images": Array [
          Object {
            "height": 500,
            "url": "test image url",
            "width": 100,
          },
        ],
        "isOpen": false,
        "name": "Artist 1",
        "owner": Object {
          "id": "testOwner",
          "type": "testType",
        },
        "public": true,
        "tracks": Object {
          "list": Array [
            Object {
              "album": Object {
                "album_type": "test type",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "images": Array [
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                ],
                "name": "Test album",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "id": "324",
                  "name": "Test Artist",
                },
              ],
              "duration_ms": 12323,
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "1234",
              "popularity": 78,
              "track_number": 1,
            },
            Object {
              "album": Object {
                "album_type": "test type",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "images": Array [
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                ],
                "name": "Test album",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "id": "324",
                  "name": "Test Artist",
                },
              ],
              "duration_ms": 12323,
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "1234",
              "popularity": 78,
              "track_number": 1,
            },
            Object {
              "album": Object {
                "album_type": "test type",
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "images": Array [
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                  Object {
                    "height": 500,
                    "url": "test image url",
                    "width": 100,
                  },
                ],
                "name": "Test album",
              },
              "artists": Array [
                Object {
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "id": "324",
                  "name": "Test Artist",
                },
              ],
              "duration_ms": 12323,
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "1234",
              "popularity": 78,
              "track_number": 1,
            },
          ],
        },
        "type": "testType",
      },
      "playlistContainsThisTrack": false,
      "removePlaylistTrackFromFinal": [Function],
      "track": Object {
        "album": Object {
          "album_type": "test type",
          "artists": Array [
            Object {
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "324",
              "name": "Test Artist",
            },
          ],
          "external_urls": Object {
            "spotify": "test_external_url",
          },
          "images": Array [
            Object {
              "height": 500,
              "url": "test image url",
              "width": 100,
            },
            Object {
              "height": 500,
              "url": "test image url",
              "width": 100,
            },
          ],
          "name": "Test album",
        },
        "artists": Array [
          Object {
            "external_urls": Object {
              "spotify": "test_external_url",
            },
            "id": "324",
            "name": "Test Artist",
          },
        ],
        "duration_ms": 12323,
        "external_urls": Object {
          "spotify": "test_external_url",
        },
        "id": "1234",
        "popularity": 78,
        "track_number": 1,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addErrorToApp": [Function],
        "addPlaylistTrackToFinal": [Function],
        "classes": Object {
          "mediaPlayer": "Track-mediaPlayer-4",
          "preview": "Track-preview-3",
          "trackInfo": "Track-trackInfo-2",
          "trackInfoContainer": "Track-trackInfoContainer-1",
        },
        "index": 1,
        "navigation": Object {
          "currentPage": "/app",
          "index": 0,
          "indexToRouteMap": Object {
            "0": "/app",
            "1": "/app/public",
            "2": "/app/componofy",
          },
          "nextIndex": 1,
          "nextPage": "/app/public",
          "numberOfPages": 3,
          "routeToIndexMap": Object {
            "/app": 0,
            "/app/componofy": 2,
            "/app/public": 1,
          },
        },
        "playlist": Object {
          "external_urls": Object {
            "spotify": "test_external_url",
          },
          "href": "testLink",
          "id": "8752",
          "images": Array [
            Object {
              "height": 500,
              "url": "test image url",
              "width": 100,
            },
          ],
          "isOpen": false,
          "name": "Artist 1",
          "owner": Object {
            "id": "testOwner",
            "type": "testType",
          },
          "public": true,
          "tracks": Object {
            "list": Array [
              Object {
                "album": Object {
                  "album_type": "test type",
                  "artists": Array [
                    Object {
                      "external_urls": Object {
                        "spotify": "test_external_url",
                      },
                      "id": "324",
                      "name": "Test Artist",
                    },
                  ],
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "images": Array [
                    Object {
                      "height": 500,
                      "url": "test image url",
                      "width": 100,
                    },
                    Object {
                      "height": 500,
                      "url": "test image url",
                      "width": 100,
                    },
                  ],
                  "name": "Test album",
                },
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "duration_ms": 12323,
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "id": "1234",
                "popularity": 78,
                "track_number": 1,
              },
              Object {
                "album": Object {
                  "album_type": "test type",
                  "artists": Array [
                    Object {
                      "external_urls": Object {
                        "spotify": "test_external_url",
                      },
                      "id": "324",
                      "name": "Test Artist",
                    },
                  ],
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "images": Array [
                    Object {
                      "height": 500,
                      "url": "test image url",
                      "width": 100,
                    },
                    Object {
                      "height": 500,
                      "url": "test image url",
                      "width": 100,
                    },
                  ],
                  "name": "Test album",
                },
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "duration_ms": 12323,
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "id": "1234",
                "popularity": 78,
                "track_number": 1,
              },
              Object {
                "album": Object {
                  "album_type": "test type",
                  "artists": Array [
                    Object {
                      "external_urls": Object {
                        "spotify": "test_external_url",
                      },
                      "id": "324",
                      "name": "Test Artist",
                    },
                  ],
                  "external_urls": Object {
                    "spotify": "test_external_url",
                  },
                  "images": Array [
                    Object {
                      "height": 500,
                      "url": "test image url",
                      "width": 100,
                    },
                    Object {
                      "height": 500,
                      "url": "test image url",
                      "width": 100,
                    },
                  ],
                  "name": "Test album",
                },
                "artists": Array [
                  Object {
                    "external_urls": Object {
                      "spotify": "test_external_url",
                    },
                    "id": "324",
                    "name": "Test Artist",
                  },
                ],
                "duration_ms": 12323,
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "id": "1234",
                "popularity": 78,
                "track_number": 1,
              },
            ],
          },
          "type": "testType",
        },
        "playlistContainsThisTrack": false,
        "removePlaylistTrackFromFinal": [Function],
        "track": Object {
          "album": Object {
            "album_type": "test type",
            "artists": Array [
              Object {
                "external_urls": Object {
                  "spotify": "test_external_url",
                },
                "id": "324",
                "name": "Test Artist",
              },
            ],
            "external_urls": Object {
              "spotify": "test_external_url",
            },
            "images": Array [
              Object {
                "height": 500,
                "url": "test image url",
                "width": 100,
              },
              Object {
                "height": 500,
                "url": "test image url",
                "width": 100,
              },
            ],
            "name": "Test album",
          },
          "artists": Array [
            Object {
              "external_urls": Object {
                "spotify": "test_external_url",
              },
              "id": "324",
              "name": "Test Artist",
            },
          ],
          "duration_ms": 12323,
          "external_urls": Object {
            "spotify": "test_external_url",
          },
          "id": "1234",
          "popularity": 78,
          "track_number": 1,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
